src/main/java/duke/task/DukeList.java

```diff
@@ -0,0 +1,233 @@
+     * @param body Description of the task to be loaded.
+     * @throws DukeException When unknown task type is loaded.
+     */
+    public void loadData(String type, String state, String body) throws DukeException {
+        Task task;
+        String[] sections;
+
+        switch (type) {
+        case "todo":
+            task = new ToDo(body);
+            break;
+        case "deadline":
+            sections = body.split(" /by ", 2);
+            task = new Deadline(sections[0], sections[1]);
+            break;
+        case "event":
+            sections = body.split(" /at ", 2);
+            task = new Event(sections[0], sections[1]);
+            break;
+        default:
+            throw new DukeException("☹ OOPS!!! Unknown task type in saved data");
+        }
+
+        if (state.equals("1")) {
+            task.setDone();
+        }
+
+        this.tasks.add(task);
+    }
+
+    /**
+     * Adds a ToDo task to the list.
+     *
+     * @param text Body of the duke.task to be added.
+     * @return The message associated to the task being displayed.
+     * @throws DukeException If there is no text.
+     */
+    public String addToDo(String text) throws DukeException {
+        String message = text.trim();
+
+        if (message.equals("")) {
+            throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+
+        ToDo input = new ToDo(message);
+
+        tasks.add(input);
+
+        return displayTask(input);
+    }
+
+    /**
+     * Adds a Deadline task to the list.
+     *
+     * @param text Body of the task to be added.
+     * @return The message associated to the task being displayed.
+     */
+    public String addDeadline(String text) throws DukeException {
+        String[] strings = text.split(" /by ", 2);
+
+        String limit = strings[1];
+        try {
+            Deadline input = new Deadline(strings[0].trim(), limit);
+
+            tasks.add(input);
+
+            return displayTask(input);
+        } catch (DateTimeParseException e) {
+            throw new DukeException("☹ OOPS!!! The deadline follows the format yyyy-MM-dd.");
+        }
+    }
+
+    /**
+     * Adds an Event task to the list.
+     *
+     * @param text Body of the task to be added.
+     * @return The message associated to the task being displayed.
+     */
+    public String addEvent(String text) {
+        String[] strings = text.split(" /at ", 2);
+
+        String limit = strings.length == 1 ? "" : strings[1];
+
+        Event input = new Event(strings[0].trim(), limit);
+
+        tasks.add(input);
+
+        return displayTask(input);
+    }
+
+    /**
+     * Lists out the current tasks in the list.
+     *
+     * @return The list of tasks in the list.
+     */
+    public String list() {
+        String str = "Here are the tasks in your list:\n";
+        for (int i = 0; i < tasks.size(); i++) {
+            str += (i + 1) + "." + tasks.get(i).toString() + "\n";
```

Comment:
Try using StringBuilder for concatenating long strings like this? If list of tasks is 100 tasks long, iirc using + requires a lot of computing power or smth